packages/jest-circus/src/run.js

```diff
@@ -64,6 +64,10 @@ const _runTestsForDescribeBlock = async (describeBlock: DescribeBlock) => {
     let numRetriesAvailable = retryTimes;
 
     while (numRetriesAvailable > 0 && test.errors.length > 0) {
+      // Clear errors so retries occur
+      // TODO: consider creating test.hookErrors and test.errors
+      test.errors = [];
```

Comment:
It would be nice to keep all errors and not just the last (in case it still errors after retrying). Thoughts?

Reply:
Given that you will still see output for each test (if your test throws an error or something) I think the user will have enough information already in their console to be able to diagnose problems.

Reply:
the only thing i'm worried about is modifying the data structure directly instead of dispatching an action (cause FLUX!)

i sense the possibility of `wtf did my errors disappear??` in future development :)

Reply:
Yeah it's ultimately necessary, though, due to the checks performed in jest-circus. We need to synchronously update those errors in this case otherwise the test isn't actually re-run.

Reply:
If you have an architectural suggestion I'm all for it :) But right now the feature doesn't work as advertised.

Reply:
are you talking about the line 104 in this file?

`dispatch` actually works synchronously, so if you just dispatch an action and update `test` from the `event_handler` it should be fine (cause it's still the same reference)

Reply:
i kind of want to experiment with redux and immutablejs and see if it affects performance. just to make sure all data structures are safe and don't get mutated in utility functions

Reply:
Ahhhh ok - let me try that!